common/models/user.js

```diff
@@ -434,6 +441,126 @@ module.exports = function(User) {
   };
 
   /**
+   * Returns default verification options to use when calling User.verify()
+   * without the `verifyOptions` param. This is specifically useful when calling
+   * this method remotely over rest.
+   *
+   * This is the full list of possible params, with example values
+   *
+   * ```js
+   * {
+   *   type: 'email',
+   *   mailer: anEmailModel,
+   *   to: 'test@email.com',
+   *   from: 'noreply@email.com'
+   *   subject: 'verification email subject',
+   *   text: 'verification email text',
+   *   headers: {'Mime-Version': '1.0'},
+   *   template: 'path/to/template.ejs',
+   *   redirect: '/',
+   *   verifyHref: '',
+   *   host: 'localhost'
+   *   protocol: 'http'
+   *   port: 3000,
+   *   restApiRoot= '/api',
+   *   tokenGenerator: function (user, cb) { cb("random-token"); }
+   * }
+   * ```
+   *
+   * The default options can be modified in your custom user model like follows:
+   *
+   * ```js
+   * customUserModel.getVerifyOptions = function() {
+   *   const base = MyUser.base.getVerifyOptions();
+   *   return Object.assign({}, base, {
+   *     // custom values
+   *   });
+   * }
+   * ```
+   *
+   * Usually you should only require to modify a subset of these params
+   * See `User.verify()` and `User.prototype.verify()` doc for params reference
+   */
+
+  User.getVerifyOptions = function() {
+    return this.settings.verifyOptions || {type: 'email'};
+  };
```

Comment:
the only missing param required to make the feature run OOTB is email `from`
maybe we should default it to something like 'noreply@noreply.com' in order to have OOTB support

also, based on our previous discussions, i implemented this with a possibility to leverage on this.settings.verifyOptions, but also described extensibility at method level.

i think it's fine to keep both in the doc, although extending via settings will not support the tokenGenerator param which expects a function. thoughts?